frame/utility/src/lib.rs

```diff
@@ -109,13 +109,33 @@ pub mod pallet {
 		BatchCompleted,
 	}
 
+	#[pallet::extra_constants]
+	impl<T: Config> Pallet<T> {
+		/// The limit on the number of batched calls.
+		fn batched_calls_limit() -> u32 {
+			let allocator_limit = 33554432;
```

Comment:
```suggestion
			let allocator_limit = 33554432;
```

feels this number should come from somewhere else?

Reply:
indeed but we can't currently link to it because it is in the client.
Maybe the proper solution is to have a configuratble limit: `type MaxBatched: Get<u32>`, or move the allocator limit to the primitives

Reply:
maybe better to set a hardcoded limit of 10_000 and then assert this in a test

Reply:
I think it is safe to merge like this because the allocator limit should not decrease, that said I opened an issue https://github.com/paritytech/substrate/issues/9392 and a PR https://github.com/paritytech/substrate/pull/9393
once merge I'll update this code as well.